File: modules/handlers/help.py
"""
:project: telegram-onedrive
:author: L-ING
:copyright: (C) 2023 L-ING <hlf01@icloud.com>
:license: MIT, see LICENSE for more details.
"""

from telethon import events
from modules.client import tg_bot
from modules.env import tg_user_name
from modules.global_var import help_res
from modules.utils import check_in_group


@tg_bot.on(events.NewMessage(pattern="/help", incoming=True, from_users=tg_user_name))
@check_in_group
async def help_handler(event):
    """Send a message when the command /help is issued."""
    await event.respond(help_res)
    raise events.StopPropagation

File: src/handlers/help.rs
/*
:project: telegram-onedrive
:author: L-ING
:copyright: (C) 2024 L-ING <hlf01@icloud.com>
:license: MIT, see LICENSE for more details.
*/

use super::docs::format_help;
use crate::{error::Result, message::TelegramMessage, state::AppState};
use grammers_client::InputMessage;
use proc_macros::{add_context, add_trace};

pub const PATTERN: &str = "/help";

#[add_context]
#[add_trace]
pub async fn handler(message: TelegramMessage, _state: AppState) -> Result<()> {
    message
        .respond(InputMessage::html(format_help(PATTERN)))
        .await?;

    Ok(())
}
